sql/atxs/atxs.go

```diff
@@ -401,15 +401,15 @@ func AddGettingNonce(db sql.Executor, atx *types.VerifiedActivationTx) (*types.V
 		if err == nil {
 			err = add(db, atx, &nonce)
 			if err != nil {
-				return nil, err
+				return &nonce, err
```

Comment:
Why do we need to return the nonce if there was an error?

Reply:
It's not an "error" if the ATX already existed in the DB. I could add a check here `if err is sql.ErrObjectExists { return &nonce, err} else {return nil, err}` but I figured that's complicating the code without gain.

Without this change, a test that stores the same ATX twice (and expects no error) fails (because "nonce is not found").

Reply:
> a test that stores the same ATX twice (and expects no error) fails

Imo such a test should fail, maybe with a different error but trying to add the same ATX (i.e. with the same ID) to the database twice should cause an error.